docs/yaml/objects/compiler.yaml

```diff
@@ -617,3 +617,7 @@ methods:
       type: list[str]
       description: |
         Extra flags to pass to the preprocessor
+    dependencies:
+      type: dep | list[dep]
```

Comment:
We don't generally call out `T | array[T]` (including right above this), because listification is just part of Meson

Reply:
That's how its defined in the common `_dependencies` declaration that I had to copy/paste because it has a different `since` value. They are all like that in this file.